components/sensor/veml7700.rst

```diff
@@ -0,0 +1,203 @@
+`DFRobot`_, and others.
+
+.. _Adafruit: http://www.adafruit.com/products/4162
+.. _SparkFun: https://www.sparkfun.com/products/15436
+.. _DFRobot: https://www.dfrobot.com/product-1620.html
+
+
+.. figure:: images/veml7700-full.jpg
+    :align: center 
+    :width: 70.0%
+
+    VEML7700 Ambient Light Sensor on a board and standalone sensors.
+
+.. figure:: images/veml7700-ui.png
+    :align: center
+    :width: 60.0%
+
+    VEML sensors in Home Assistant UI.
+
+The sensor is a high accuracy ambient light digital 16-bit resolution sensor with dynamic range from 0 lux to about 120,000 lux.
+Wide range of measurements enabled by four configurable *gain* levels and six different *integration time* options.
+Higher values used for lower light conditions. 
+
+The sensor has two photodiodes with different spectral response represented by two channels: *ALS* channel and *WHITE* channel.
+*ALS*, or *Ambient light* channel follows so-called human eye curve very well. *WHITE* channel offers a much higher responsivity 
+for a much wider wavelength spectrum, capturing quite a lot of near-infrared radiance.
+
+.. figure:: images/veml7700-spectral.png
+    :align: center
+    :width: 100.0%
+
+    ALS and WHITE channels spectral response
+
+
+Using Automatic measurement mode is advised: proper *gain* and *integration time* are selected by the component by doing several 
+measurements, as per recommendations by manufacturer.
+
+Should you desire to manually control those parameters - please note that:
+
+- Gain levels of 1/8 and 1/4 shall be used in most cases. 
+- Gain levels of 1 an 2 are only intended to be used in low light conditions < 100 lux. For very high illuminations it shows high non-linearity.
+- In the range of 0 lux - 1000 lux the sensor measurements are stricly linear for Gain 1/4 and 1/8, after 1000 lux it shows non-linearity. 
+
+
+Lux compensation formula is used to get better readings in bright conditions. 
+However, it gives quite high error in very bright direct sunlight (instead of 100-120 kilolux it might give 150-200k+).
+
+This Wikipedia `article <https://en.wikipedia.org/wiki/Lux>`__ has a table of some lux values for comparison.
+
+Automatic measurement mode
+--------------------------
+
+In automatic measurement mode the component starts from Gain 1/8 and 100 ms. 
+
+- If illuminance is higher than 46 lx (100 counts) it gradually reduces integration time to get good reading (best in range 100 - 10,000 counts). 
+- In case of low illuminance (less than 46 lx) it tries to gradually increase gain and only then increase integration time.
+
+Please note, that in low light conditions measurement process might take several seconds due to long exposure periods and sensor reconfigurations.
+
+Lux compensation
+----------------
+
+Lux compensation is done as per recommendation from the manufacturer, however it can be turned off setting ``lux_compensation: false``
+in configuration.
```

Comment:
```suggestion
Lux compensation is done as recommended by the manufacturer, however it can be turned off by
setting ``lux_compensation: false`` in your device's configuration.
```